src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/oops/ConstMethod.java

```diff
@@ -666,4 +621,67 @@ private long offsetOfExceptionTable() {
     return offset;
   }
 
+  private boolean hasMethodParameters() {
+    return (getFlags() & HAS_METHOD_PARAMETERS) != 0;
+  }
+
+  private boolean hasGenericSignature() {
+    return (getFlags() & HAS_GENERIC_SIGNATURE) != 0;
+  }
+
+  private Address getAddressAtOffset(int offsetInWords) {
+    return getAddress().getAddressAt(offsetInWords * VM.getVM().getAddressSize());
+  }
+
+  // Pointers to annotations are stored towards the end of the ConstMethod in following format:
+  //
+  //   start of ConstMethod                          end of ConstMethod
+  //            |                                           |
+  //            V                                           V
+  //            | ... | default | type | parameter | method |
```

Comment:
So the `...` part is represented by `getSize()`? It would be good to call that out. Also point out that each of the annotations pointers is optional.

Reply:
Updated the comment.